include/knowhere/kmeans.h

```diff
@@ -0,0 +1,157 @@
+// Copyright (C) 2019-2024 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#pragma once
+
+#include <cstdlib>
+#include <fstream>
+#include <string_view>
+#include <vector>
+
+#include "knowhere/config.h"
+#include "knowhere/dataset.h"
+#include "knowhere/expected.h"
+#include "knowhere/file_manager.h"
+
+namespace knowhere::kmeans {
+namespace {
+
+static constexpr int64_t MAX_TRAIN_SIZE = 10000000L * 700 * 4;
+
+template <typename T>
+inline bool
+load_bin_file(const std::string& fname, std::unique_ptr<T[]>& data, uint64_t& offset) {
```

Comment:
do I get it correct that this function loads the whole file into a given preallocated buffer into a given offset? If so, then could you please add a comment on what this function does exactly, otherwise it is somewhat confusing. Thanks.

Reply:
updated